kbase-extension/static/kbase/js/widgets/function_output/kbaseFeatureSet.js

```diff
@@ -62,162 +64,270 @@ define (
+                                                feature_id,
+                                                aliases: {na: 'na'},
+                                                feature_type: 'na',
+                                                function: 'na'
+                                            };
+                                        }));
+                                }
+                            })(),
+                            objectRef,
+                            typeName,
+                            info
+                        ]);
+                    }));
+                })
+                .then((result) => {
+                    for (const [features, objectRef, objectType, objectInfo] of result) {
+                        const objectName = objectInfo[1];
+                        for (const feature of features) {
+                            this.featureTableData.push(
                                 {
-                                    fid: '<a href="/#dataview/'+gid+
-                                                '?sub=Feature&subid='+feature.feature_id + '" target="_blank">'+
-                                                feature.feature_id+'</a>',
-                                    gid: '<a href="/#dataview/'+gid+
-                                            '" target="_blank">'+gid+"</a>",
-                                    ali: Object.keys(feature.aliases).join(", "),
+                                    fid: '<a href="/#dataview/'+objectRef+
+                                                            '?sub=Feature&subid='+feature.feature_id + '" target="_blank">'+
+                                                            feature.feature_id+'</a>',
+                                    objectType,
+                                    gid: '<a href="/#dataview/'+objectRef+
+                                                        '" target="_blank">'+objectName+'</a>',
+                                    aliases: Object.keys(feature.aliases).join(', '),
                                     type: feature.feature_type,
                                     func: feature.function
                                 }
                             );
-
                         }
-                        self.renderFeatureTable(); // just rerender each time
-                        self.loading(true);
-                    })
-                    .fail(function(e) {
-                        console.error(e);
-                    });
-            }
-            if (Object.keys(self.features).length === 0){
-                self.loading(true);
-                self.showMessage("This feature set is empty.")
-            }
+                    }
+                })
+                .then(() => {
+                    this.loading(false);
+                    this.renderFeatureTable(); // just rerender each time
+                });
+
+            // return Promise.all(
```

Comment:
Is this all needed still? Can it just be removed?

Reply:
I think a lot of commented out code has been removed.